magenta/interfaces/midi/midi_hub.py

```diff
@@ -148,85 +149,84 @@ class Metronome(threading.Thread):
     velocity: The velocity of the metronome's tick `note_on` message.
     pitch: The pitch of the metronome's tick `note_on` message.
     duration: The duration of the metronome's tick.
-    qpb: The number of quarter notes per bar.
   """
+  # The pitch interval in semitones to raise metronome downbeats only.
+  _ACCENT_PITCH_INTERVAL = 7
+  # The amount to add to metronome velocity on downbeats only.
+  _ACCENT_VELOCITY_DELTA = 1
 
   def __init__(self,
                outport,
                qpm,
+               qpb,
                start_time,
                stop_time=None,
                velocity=_DEFAULT_METRONOME_VELOCITY,
                pitch=_DEFAULT_METRONOME_PITCH,
-               duration=_DEFAULT_METRONOME_TICK_DURATION,
-               qpb=4):
+               duration=_DEFAULT_METRONOME_TICK_DURATION):
     self._outport = outport
     self._qpm = qpm
+    self._qpb = qpb
     self._start_time = start_time
     self._velocity = velocity
     self._pitch = pitch
     self._duration = duration
     # A signal for when to stop the metronome.
     self._stop_time = stop_time
-    self._qpb = qpb
-    self._count = 0
-    # The amount to add to metronome velocity on downbeats only.
-    self._accent_velocity_delta = 10
-    # The pitch interval in semitones to raise metronome downbeats only.
-    self._accent_pitch_interval = 7
     super(Metronome, self).__init__()
 
-    # limit metronome accent velocity to 127 max value.
-    self._accent_velocity = min(velocity + self._accent_velocity_delta, 127)
-    # only raise metronome accent pitch if there is headroom.
-    self._accent_pitch = (
-        self._pitch if self._pitch + self._accent_pitch_interval > 127
-        else self._pitch + self._accent_pitch_interval)
 
   def run(self):
     """Outputs metronome tone on the qpm interval until stop signal received."""
+
+    # limit metronome accent velocity to 127 max value.
+    accent_velocity = min(self._velocity + self._ACCENT_VELOCITY_DELTA, 127)
+    # only raise metronome accent pitch if there is headroom.
+    accent_pitch = (
+      self._pitch if self._pitch + self._ACCENT_PITCH_INTERVAL > 127
+      else self._pitch + self._ACCENT_PITCH_INTERVAL)
+
     period = 60. / self._qpm
+    metric_position = 1
     sleeper = concurrency.Sleeper()
     now = time.time()
     next_tick_time = max(
         self._start_time,
         now + period - ((now - self._start_time) % period))
     while self._stop_time is None or self._stop_time > next_tick_time:
-      self._count += 1
       sleeper.sleep_until(next_tick_time)
 
-      if self._count % self._qpb == 1:
+      if metric_position % self._qpb == 1:
```

Comment:
In this conditional, just set a pitch and velocity and move the sends outside of the conditional